### PR TITLE
SCHED-196: Modifies Process Manager

### PR DESCRIPTION
**Pull request by stroncod**
Adds Singleton class for all our singletons! (ProcessManager y PlanManager)

### CODE REVIEW

File: graphql_server/__init__.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 from copy import deepcopy
 from datetime import date, datetime
-from threading import Lock
+from tokenize import Single
```
Comment (sraaphorst) [inline graphql_server/__init__.py:3]: What does this do and where is it used? I don't see any reference to `Single` in the code.

File: graphql_server/__init__.py [hunk 2]
```diff
@@ -85,22 +86,15 @@ def for_site(self, site: Site) -> 'SPlans':
         )
 
 
-class PlanManager:
+class PlanManager(metaclass=Singleton):
```
Comment (sraaphorst) [inline graphql_server/__init__.py:89]: Do we really need metaclasses? Can't we perhaps do something simpler?

https://realpython.com/python-metaclasses

File: graphql_server/__init__.py (hunk 1 above)
Comment (stroncod) [reply graphql_server/__init__.py:3]: VS Code add some dumb stuff to autocomplete. Thanks, I'll delete it

File: graphql_server/__init__.py (hunk 2 above)
Comment (stroncod) [reply graphql_server/__init__.py:88]: A Base class can get the same results but metaclass is the recommended singleton implementation: https://stackoverflow.com/questions/6760685/creating-a-singleton-in-python

Comment (sraaphorst) [reply graphql_server/__init__.py:88]: Okay, sounds good.

**Review by sraaphorst — APPROVED** [review]

File: graphql_server/__init__.py (hunk 1 above)
Comment (sraaphorst) [reply graphql_server/__init__.py]: LOL happens to all of us sometimes.

**Pull request merged**